tests/contrib/tracking/test_hashing.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import absolute_import, division, print_function
+
+import pytest
+import torch
+
+from pyro.contrib.tracking.hashing import LSH, ApproxSet
+from tests.common import assert_equal
+
+
+@pytest.mark.parametrize('scale', [-1., 0., -1 * torch.ones(2, 2)])
+def test_lsh_init(scale):
+    num_error = 0
+    try:
+        LSH(scale)
+    except AssertionError:
+        num_error += 1
+    else:
+        pass
+    finally:
+        assert num_error > 0
+
+
+@pytest.mark.parametrize('scale', [0.1, 1, 10, 100])
+def test_lsh_add(scale):
+    lsh = LSH(scale)
+    a = torch.rand(10)
+    lsh.add('a', a)
+    assert lsh._hash_to_key[lsh._key_to_hash['a']] == {'a'}
+
+
+@pytest.mark.parametrize('scale', [0.1, 1, 10, 100])
+def test_lsh_hash_nearby(scale):
+    k = 10
```

Comment:
Lookup has complexity `O(3^k)`. If these tests are slow, try reducing k to say 7.

Reply:
BTW we could easily replace this implementation with an `O(2^k)` version which is much cheaper for higher dimensions.